frontend/src/pages/projects/screens/detail/storage/AddStorageModal.tsx

```diff
@@ -3,23 +3,55 @@ import { Button, Modal, Stack, StackItem } from '@patternfly/react-core';
 import CreateNewStorageSection from './CreateNewStorageSection';
 import AddExistingStorageSection from './AddExistingStorageSection';
 import { useStorageDataObject } from '../../spawner/storage/utils';
+import { checkRequiredFieldsForAddingStorage } from './utils';
+import { assemblePvc, createPvc } from '../../../../../api';
+import { ProjectDetailsContext } from '../../../ProjectDetailsContext';
 
 import './addStorageModal.scss';
 
 type AddStorageModalProps = {
   isOpen: boolean;
   onClose: () => void;
```

Comment:
You'll probably need to pass a boolean value to this modal to say if it closed with submit or not. (to satisfy my other comment)